Twittimation/Optional.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+
+public struct Optional<T>
+{
+    public bool HasValue { get; private set; }
+    private T value;
+    public T Value
+    {
+        get
+        {
+            if (!HasValue)
+                throw new InvalidOperationException();
+            return value;
+        }
+    }
+
+    public Optional(T value)
+    {
+        this.value = value;
+        HasValue = true;
+    }
+
+    public static implicit operator Optional<T>(T value)
+    {
+        return new Optional<T>(value);
+    }
+
+    public static implicit operator Optional<T>(Optional<ushort> v)
+    {
+        throw new NotImplementedException();
```

Comment:
Is this needed?

Reply:
Hmm, wonder why I had that code. It will be removed.